src/submission/forms.py

```diff
@@ -238,13 +241,18 @@ class ArticleInfoSubmit(ArticleInfo):
     # Filter licenses and sections to publicly available only
     FILTER_PUBLIC_FIELDS = True
 
+    def __init__(self, *args, **kwargs):
+        super(ArticleInfoSubmit, self).__init__(*args, **kwargs)
+        self.fields.pop('competing_interests')
+
 
 class EditorArticleInfoSubmit(ArticleInfo):
     # Used when an editor is making a submission.
     FILTER_PUBLIC_FIELDS = False
 
     def __init__(self, *args, **kwargs):
         super(EditorArticleInfoSubmit, self).__init__(*args, **kwargs)
+        self.fields.pop('competing_interests')
```

Comment:
Barbara Liskov wouldn't be proud of this, can we instead not have `competing_interests` on the base form and have a different child when it is needed?

I think what we meant to do here is to create a new `EditArticleMetadata` from that adds whichever fields are needed (in this case `competing_interests`)

```mermaid
classDiagram
    class ArticleInfo {
        <<abstract>>
    }

    class ArticleInfoSubmit {
        
    }
    class EditorArticleInfoSubmit {
        
    }
    class EditArticleMetadata {
competing_interests: String
}

    ArticleInfo <|-- ArticleInfoSubmit
    ArticleInfo <|-- EditorArticleInfoSubmit
    ArticleInfo <|-- EditArticleMetadata
    ```